### DISCUSSION TITLE
【v5】tooltip position auto 时，鼠标 hover 异常

### DISCUSSION THREAD

**Discussion by Deathsteps** [antvis/G2#5163] [Q&A] [answered]
## 问题描述

<img width="1222" alt="image" src="https://github.com/antvis/G2/assets/1549118/cbf0d91f-65f3-41fe-a732-36c24d58e3c3">

## 期望结果

position auto, 鼠标 hover 显示正常

## 如何重现

官方代码修改
```javascript
/**
 * A recreation of this demo: https://observablehq.com/@d3/bar-chart
 */
import { Chart } from '@antv/g2';

const chart = new Chart({
  container: 'container',
  theme: 'classic',
  autoFit: true,
});

chart
  .interval()
  .data({
    type: 'fetch',
    value:
      'https://gw.alipayobjects.com/os/bmw-prod/fb9db6b7-23a5-4c23-bbef-c54a55fee580.csv',
  })
  .encode('x', 'letter')
  .encode('y', 'frequency')
  .axis('y', { labelFormatter: '.0%' });

chart.interaction('tooltip', { position: 'auto' });

chart.render();
```

## 额外信息
* G2 5.0.11

**Comment by hustcc** [comment]
tooltip.position 已经没有 auto 这个选项，只有具体的方位。你是在哪个文档看到这个配置吗？我修正一下文档。

**Reply by Deathsteps** [reply]
我现去找，找不到了。可能是之前的文档。以后不打算开 auto 这个配置吗？

**Reply by pearmini** [reply, marked as answer]
目前是不打算开 auto 这个配置了。之前的 api 设计是存在一些问题的：

- position 为非 auto 的时候，指配置 tooltip 显示的位置是在鼠标点的方位；
- position 为 auto 的时候，指 tooltip 在超出 bounds 范围的时候，其位置相对于鼠标点的方位；

这个就把两个功能耦合在一个配置中了，且带来一个问题就是 position 为 auto 且 tooltip 不超出 bounds 范围时，tooltip 的方位如何设置？这个是无法设置的。


现在的设计是分成两个：

- 在设置 bounds 后，会自动在这个范围内 auto，如果 bounds 不设置，意思就是不限制，也不会自动做 auto
- position 只是用来设置 tooltip 显示的位置是在鼠标点的方位

这样功能和配置解耦，且灵活组合，满足所有情况。
